lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1960,6 +1960,43 @@ static SubstitutionMap getSubstitutionMap(
   return substMap;
 }
 
+/// Finds the `[differentiable]` attribute on the specified original function
```

Comment:
Todo: fix doc comment and rename function to `lookUpMinimalDifferentiableAttrAndIndexSubset`.